libcob/intrinsic.c

```diff
@@ -1282,7 +1282,7 @@ cob_intr_date_of_integer (cob_field *srcdays)
 			}
 		}
 	}
-	snprintf (buff, 15, "%4.4d%2.2d%2.2d", baseyear, i, days);
+	snprintf (buff, 25, "%4.4d%2.2d%2.2d", baseyear, i, days);
```

Comment:
Even if the second argument is 24, the warning messages will be solved.
Set the second argument to 24.